v1_1/check.go

```diff
@@ -47,6 +63,23 @@ func (ch *Check) check() (string, error) {
 		}
 		resultCh := make(chan result)
 		go func() {
+
+			// Any panics hit during checking should cause the check to fail
+			defer func() {
+				var err error
+				if rec := recover(); rec != nil {
+					switch t := rec.(type) {
+					case string:
+						err = errors.New(t)
+					case error:
+						err = t
+					default:
+						err = errors.New("Unknown error")
+					}
+				}
+				resultCh <- result{"", err}
```

Comment:
Shouldn't this be inside the `if`? - otherwise you are posting two objects into the channel, one from the main func and one from this defer

Reply:
Ah, good spot!